### PR TITLE
feat: implement abbreviated manifest

### PR DESCRIPTION
**Pull request by juanpicado**
feat: implement abbreviated manifest

Enable abbreviated manifest data by adding the header:

```
curl -H "Accept: application/vnd.npm.install-v1+json" https://registry.npmjs.org/verdaccio
```

It returns a filtered manifest, additionally includes the [time](https://github.com/pnpm/rfcs/pull/2) field by request.

Current support for packages managers:

- npm: yes
- pnpm: yes
- yarn classic: yes
- yarn modern (+2.x): [no](https://github.com/yarnpkg/berry/pull/3981#issuecomment-1076566096)

https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#abbreviated-metadata-format

Follow up of https://github.com/verdaccio/verdaccio/pull/3322
Fixes for 5.x https://github.com/verdaccio/verdaccio/issues/2792

### CODE REVIEW

**Comment by juanpicado** [comment]
@zkochan I'm shipping this on `v5.15.0` with the time field included.

**Comment by zkochan** [comment]
oh, that's great!

**Comment by zkochan** [comment]
@juanpicado looks like you do not pass all the needed fields in the abbreviated version. I just tried it in the pnpm e2e tests and some tests failed https://github.com/pnpm/pnpm/runs/8172441979?check_suite_focus=true

do you include the os and cpu fields?

File: src/lib/storage-utils.ts [hunk 1]
```diff
@@ -236,3 +236,44 @@ export function isPublishablePackage(pkg: Package): boolean {
 
   return _.includes(keys, 'versions');
 }
+
+export function convertAbbreviatedManifest(manifest: Manifest): AbbreviatedManifest {
+  const abbreviatedVersions = Object.keys(manifest.versions).reduce((acc: AbbreviatedVersions, version: string) => {
+    const _version = manifest.versions[version];
+    // This should be align with this document
+    // https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#abbreviated-version-object
```
Comment (zkochan) [inline src/lib/storage-utils.ts:244]: looks like that document is incorrect. os and cpu are also included in the abbreviated version

**Pull request merged**